test/assertions/to-have-properties.spec.js

```diff
@@ -289,4 +289,58 @@ describe('to have properties assertion', () => {
       );
     });
   });
+
+  describe('with the "only" flag', () => {
+    it('should pass', () => {
+      expect(function() {
+        expect({ foo: 123, bar: 456 }, 'to only have properties', [
+          'foo',
+          'bar'
+        ]);
+      }, 'not to throw');
+    });
+
+    it('should fail with a diff and mark properties to be removed', () => {
+      expect(
+        function() {
+          expect({ foo: 123, bar: 456 }, 'to only have properties', ['foo']);
+        },
+        'to error',
+        "expected { foo: 123, bar: 456 } to only have properties [ 'foo' ]\n" +
+          '\n' +
+          '{\n' +
+          '  foo: 123,\n' +
+          '  bar: 456 // should be removed\n' +
+          '}'
+      );
+    });
```

Comment:
Would be good to also test that missing properties are correctly marked (they aren't :smirk_cat:):

```
    it('should fail with a diff and mark missing properties', () => {
      expect(
        function() {
          expect({ foo: 123 }, 'to only have properties', ['foo', 'bar']);
        },
        'to error',
        "expected { foo: 123 } to only have properties [ 'foo', 'bar' ]\n" +
          '\n' +
          '{\n' +
          '  foo: 123\n' +
          '  // missing bar\n' +
          '}'
      );
    });
```